pyne/vta.py

```diff
@@ -0,0 +1,500 @@
+"""Purpose:
+
+This module provides a modified ray traverse method, which is modifed based on
+http://www.cse.yorku.ca/~amana/research/grid.pdf, to calculate which voxels
+intersect with a given ray (line segment).
+Several rays are used to calculate the voxels that intersect with the facet.
+A series of rays are constructed based on both the facet and the mesh grid to
+make sure all the wanted voxels will be found. Combining all the found voxels
+for each ray, we get the voxels intersecting with the facet.
+
+These voxels will be used to construct the bounding volume for source sampling.
+But currently these functions are separate from MOAB. Interface of MOAB with
+PyNE is still in need.
+
+.. moduleauthor:: X. Zhang
+"""
+from __future__ import division
+import math
+import numpy as np
+
+
+##################################
+### Facet voxel traverse tools ###
+##################################
+
+
+def _is_point_in_mesh(bounds, point):
+    """
+    Check whether a point is in the mesh.
+
+    Parameters:
+    -----------
+    bounds : list
+        Boundaries. Nested list of the boundaries along x, y, and z directions.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+    point : numpy array
+        An array with size of 3, represents the point to be checked.
+
+    Return:
+    -------
+    True : when point in the mesh.
+    False : when point outside the mesh.
+    """
+    if bounds[0][0] <= point[0] <= bounds[0][-1] and \
+       bounds[1][0] <= point[1] <= bounds[1][-1] and \
+       bounds[2][0] <= point[2] <= bounds[2][-1]:
+        return True
+    else:
+        return False
+
```

Comment:
Why not check coordinates for being outside the boundaries in their respective directions?. All "AND" statements will be executed unless a "False" is found which means that all statements will be checked if they result in True. I suggest switching to "NOT OR" statements. This way if at least one coordinate is outside the mesh then there will be no need to continue checking the rest of the directions. 
Note: I might not have succeeded in explaining what I mean but you can check "short circuit" of logical operators.
```suggestion
    if not (bounds[0][0] <= point[0] <= bounds[0][-1]) or \
      not (bounds[1][0] <= point[1] <= bounds[1][-1]) or \
      not (bounds[2][0] <= point[2] <= bounds[2][-1]):
        return False
    else:
        return True
```